src/sentry/api/serializers/models/discoversavedquery.py

```diff
@@ -38,6 +45,13 @@ def serialize(self, obj, attrs, user, **kwargs):
             if obj.query.get(key) is not None:
                 data[key] = obj.query[key]
 
+        # expire queries that are beyond the retention period
+        retention = quotas.get_event_retention(organization=obj.organization)
+        if retention and "start" in obj.query:
+            start, end = parse_timestamp(obj.query["start"]), parse_timestamp(obj.query["end"])
+            start = max(start, timezone.now() - timedelta(days=retention))
+            data["expired"] = start > end
```

Comment:
@shruthilayaj It might be a good idea to have a test for this in `tests/snuba/api/endpoints/test_discover_saved_queries.py`.